src/services/schedule.ts

```diff
@@ -37,8 +41,9 @@ export interface IScheduleService {
 }
 
 export class ScheduleService implements IScheduleService {
-  public async schedule(request: IScheduleRequest, params?: IScheduleParams) {
-    if (params?.source === 'proxy') {
+  public async schedule(request: IScheduleRequest, params: IScheduleParams) {
+    const isProxyRequest = params.source === 'proxy';
+    if (isProxyRequest) {
```

Comment:
Using this instead of nested if's is a better approach? 

if (isProxyRequest && txCache.get(request.signedTransaction)) {
        throw new Error('Duplicate request');
    }
 if (isProxyRequest) {
      txCache.put(request.signedTransaction, true);
    }

Reply:
You are absolutely right, but actually the whole `schedule` function needs to be refactored to handle the various cases more clearly. I did not have a good idea how to split up the code yet, so I'm leaving the 'status quo' for now